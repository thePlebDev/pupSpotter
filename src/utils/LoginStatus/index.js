import axios from 'axios'
import {backendUrl} from '../Constants';

let logginCheck = (function(){

  return{

    checkStatus: function(setLoggedIn){
      axios.get(`${backendUrl}user/logout`)
      .then(data=>{
        if(data.data.status){
          console.log(data.data)
          setLoggedIn(false)
        }})
      .catch(e=>console.error(e))
    },

  }

}())

export default logginCheck
