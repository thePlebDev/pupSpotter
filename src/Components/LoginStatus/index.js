import React from 'react';
import {Link} from 'react-router-dom';
import PersonIcon from '@material-ui/icons/Person';
import styled from 'styled-components'
import axios from 'axios'
import PropTypes from 'prop-types';
import {backendUrl} from '../../utils/Constants'


const Text = styled.div`
  font-size:1.4em;
  color:white;
  margin-left:60px;
  cursor:pointer;
`

const LoginStatus = ({loggedIn,setLoggedIn})=>{
  const handleClick =()=>{
    axios.get(`${backendUrl}user/logout`)
    .then(data=>{
      if(data.data.status){
        console.log(data.data)
        setLoggedIn(false)
      }})
    .catch(e=>console.error(e))
  }
  return(
    <div style={{display:'flex',alignItems:'center'}} data-testid="login">
    {
      loggedIn
         ?
    <Text onClick={()=>{handleClick()}} data-testid="login">Logout</Text>
         :
    <Link to="/login">
      <Text>Login</Text>
    </Link>
    }

    {
      loggedIn
          ?
      <PersonIcon style={{fontSize:'60',color:'#5CDB95',cursor:'pointer',marginLeft:'25px' }} data-testid="login"  />
          :
      <PersonIcon style={{fontSize:'60',color:'white',cursor:'pointer',marginLeft:'40px'}}  />
    }

    </div>
  )

}

LoginStatus.propTypes ={
  loggedIn:PropTypes.bool,
  setLoggedIn:PropTypes.func
}

export default LoginStatus
