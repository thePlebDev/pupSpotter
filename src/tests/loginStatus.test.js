import React from 'react';
import LoginStatus from '../Components/LoginStatus'
import { shallow } from 'enzyme';

import EnzymeAdapter from 'enzyme-adapter-react-16';
import Enzyme from 'enzyme';
Enzyme.configure({
  adapter: new EnzymeAdapter()
})


describe('testing the LoginStatus Component',()=>{
  it('should render with loggedIn set to true',()=>{
    const wrapper = shallow(<LoginStatus loggedIn={true} />)
    expect(wrapper.find('[data-testid="login"]').length).toBe(2)
    expect(wrapper.find('[data-testid="loginButton"]').length).toBe(1)
  })
  it('should render with loggedIn set to false',()=>{
    const wrapper = shallow(<LoginStatus loggedIn={false} />)
    expect(wrapper.find('[data-testid="login"]').length).toBe(1)
    expect(wrapper.find('[data-testid="logged out"]').length).toBe(1)
  })
})
