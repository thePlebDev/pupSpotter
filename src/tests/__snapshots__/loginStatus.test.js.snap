// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the login status component setting up the snapshot 1`] = `
<div
  data-testid="login"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
    }
  }
>
  <a
    href="/login"
    onClick={[Function]}
  >
    <div
      className="sc-AxjAm eEJuQa"
    >
      Login
    </div>
  </a>
  <svg
    aria-hidden={true}
    className="MuiSvgIcon-root"
    focusable="false"
    style={
      Object {
        "color": "white",
        "cursor": "pointer",
        "fontSize": "60",
        "marginLeft": "40px",
      }
    }
    viewBox="0 0 24 24"
  >
    <path
      d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
    />
  </svg>
</div>
`;
